internal/util/get/get_test.go

```diff
@@ -751,3 +752,156 @@ func TestCommand_DefaultValues_AtVersion(t *testing.T) {
 	c = Command{Git: kptfile.Git{Repo: "foo", Directory: "/", Ref: "r"}}
 	assert.EqualError(t, c.DefaultValues(), "must specify destination")
 }
+
+func TestCommand_Run_subpackages(t *testing.T) {
+	testCases := []struct {
+		name            string
+		directory       string
+		ref             string
+		upstream        testutil.Content
+		refRepos        map[string][]testutil.Content
+		expectedResults *pkgbuilder.RootPkg
+	}{
+		{
+			name:      "basic package",
+			directory: "/",
+			ref:       "master",
+			upstream: testutil.Content{
+				Pkg: pkgbuilder.NewRootPkg().
+					WithKptfile().
+					WithResource(pkgbuilder.DeploymentResource),
+			},
+			expectedResults: pkgbuilder.NewRootPkg().
+				WithKptfile().
+				WithResource(pkgbuilder.DeploymentResource),
+		},
+		{
+			name:      "package with subpackages",
+			directory: "/",
+			ref:       "master",
+			upstream: testutil.Content{
+				Pkg: pkgbuilder.NewRootPkg().
+					WithKptfile().
+					WithResource(pkgbuilder.DeploymentResource).
+					WithSubPackages(
+						pkgbuilder.NewSubPkg("subpkg").
+							WithKptfile().
+							WithResource(pkgbuilder.ConfigMapResource),
+					),
+			},
+			expectedResults: pkgbuilder.NewRootPkg().
+				WithKptfile().
+				WithResource(pkgbuilder.DeploymentResource).
+				WithSubPackages(
+					pkgbuilder.NewSubPkg("subpkg").
+						WithKptfile().
+						WithResource(pkgbuilder.ConfigMapResource),
+				),
+		},
+		{
+			name:      "package with remote subpackages",
+			directory: "/",
+			ref:       "master",
+			upstream: testutil.Content{
+				Pkg: pkgbuilder.NewRootPkg().
+					WithKptfile(
+						pkgbuilder.NewKptfile().
+							WithSubpackages(
+								pkgbuilder.NewSubpackage("foo", "/", "master", "fast-forward"),
+							),
+					).
+					WithResource(pkgbuilder.DeploymentResource).
+					WithSubPackages(
+						pkgbuilder.NewSubPkg("subpkg").
+							WithResource(pkgbuilder.ConfigMapResource),
+					),
+			},
+			refRepos: map[string][]testutil.Content{
```

Comment:
Should we test a scenario where local and remote subpackage with same name ?

I think we can have many combination of test cases with local and remote subpackages clashes. I am fine with approving this PR as is and add more complex test cases in future.

Reply:
Yeah, this test is just an initial example. Next step is to actually update the `kpt pkg get` functionality to handle the nested packages and as part of that effort I expect to add a substantial number of tests.